tests/e2e_regression/e2e_config_1.yml

```diff
@@ -0,0 +1,42 @@
+analysis:
```

Comment:
I'm worried about the _1 and _2 config files drifting apart, leading to hard-to-understand failures in e2e tests.

One way to avoid it would be to derive them programmatically from some common template. But I think it would be relatively verbose to do it, and potentially awkward to maintain.

A more light-weight option is to add a step to the test itself that makes sure the two configs are both talking to the same cache, both expecting mainnet (vs testnet), etc. Something like this:

```bash
#!/bin/bash

set -euo pipefail

# Our e2e regression test runs the analyzer twice in close succession. This is a slightly hacky but
# simple way to ensure block analyzers run first, and non-block analyzers perform EVM queries always
# at the same height, thereby hitting the offline response cache.
#
# This script compares the key parameters of the two config files used in the two runs. If any of
# those parameters differ, it shows the diff and exits with an error.

# Element of the config files that we'll compare
important_attrs='{"cache": .analysis.source.cache.cache_dir, "chain_name": .analysis.source.chain_name, "db": .analysis.storage.endpoint}'

# A converter whose only dependency is stock python, which is likely to be preinstalled
alias yaml2json="python -c 'import sys,yaml,json; print(json.dumps(yaml.safe_load(str(sys.stdin.read()))))'" 

# Compare
cat tests/e2e_regression/e2e_config_1.yml | yaml2json | jq "$important_attrs" > /tmp/e2e_config_1.summary
cat tests/e2e_regression/e2e_config_2.yml | yaml2json | jq "$important_attrs" > /tmp/e2e_config_2.summary
diff /tmp/e2e_config_1.summary /tmp/e2e_config_2.summary || { echo "The two config files for e2e tests differ in key parameters! See diff above."; exit 1; }
```

Then in the Makefile, at the beginning of the regression test target, we just need to call it:
```Makefile
@ensure_consistent_config.sh
```

The script is untested apart from the jq part. Please rethink `important_config` with a critical eye.

Reply:
got it, thanks for the example! Added with a couple minor tweaks
- `shopt -s expand_aliases` to make the aliases work in script
- changed `python` -> `python3` to be clearer